nrf: Move linker-defined symbols to where they are used.

aykevl opened the pull request:
`gccollect.h` is not the right place to define them.

This change does not affect firmware size, and probably doesn't change the firmware at all.

aykevl committed:
nrf: Move linker-defined symbols to where they are used.

gccollect.h is not the right place to define them.

glennrub commented on ports/nrf/main.c:
```diff
@@ -91,6 +91,7 @@ void do_str(const char *src, mp_parse_input_kind_t input_kind) {
 
 extern uint32_t _heap_start;
 extern uint32_t _heap_end;
+extern uint32_t _ram_end;
```
Would this also imply that we can remove the include of "gccollect.h"?

glennrub commented on ports/nrf/modules/machine/modmachine.c:
```diff
@@ -64,6 +64,18 @@
+extern uint32_t _heap_start;
+extern uint32_t _heap_end;
+extern uint32_t _estack;
+extern uint32_t _ram_end;
```
gccollect include can probably be removed here as well.

aykevl commented:
Rebased on master and removed `gccollect.h` where not needed.
In fact, it looks like `gccollect.h` isn't needed anywhere anymore, it's only included in `gccollect.c`.

glennrub commented:
You are right, gccollect.h can be removed. Would you like to remove it before i merge it? The rest looks good to me.

glennrub commented:
In order to clean up a bit i'm closing this PR as the continuation of the nrf-port development should happen in mainline micropython repository after the merge.
